python/pyarrow/compat.py

```diff
@@ -205,16 +191,37 @@ def import_tensorflow_extension():
     if spec:
         module = importlib.util.module_from_spec(spec)
         for path in module.__path__:
-            ext = os.path.join(path, "libtensorflow_framework.so")
-            if os.path.exists(ext):
-                import ctypes
-                try:
-                    ctypes.CDLL(ext)
-                except OSError:
-                    pass
-                tensorflow_loaded = True
-                break
+            yield path
 
+def import_tensorflow_extension():
+    """
+    Load the TensorFlow extension if it exists.
+
+    This is used to load the TensorFlow extension before
+    pyarrow.lib. If we don't do this there are symbol clashes
+    between TensorFlow's use of threading and our global
+    thread pool, see also
```

Comment:
The issue is a bit more complicated than that. The problem (is my understanding) is that devtoolset-2 statically links various parts of the C and C++ runtime, including some mutable static data members which are modified by pthreads initialization. We should document this issue in some more detail somewhere so we can point packaging folks to it to see what can be done to improve the situation in the future